openhands/server/session/manager.py

```diff
@@ -11,18 +12,19 @@
 from openhands.server.session.conversation import Conversation
 from openhands.server.session.session import Session
 from openhands.storage.files import FileStore
+from openhands.utils.async_utils import call_coro_in_bg_thread
 
 
+@dataclass
 class SessionManager:
-    _sessions: dict[str, Session] = {}
+    config: AppConfig
+    file_store: FileStore
     cleanup_interval: int = 300
     session_timeout: int = 600
+    _sessions: Dict[str, Session] = field(default_factory=dict)
+    _create_conversation_tasks: Dict[str, asyncio.Task] = field(default_factory=dict)
```

Comment:
hmm I was actually hoping to get rid of these dicts. They create confusing when operating in HA mode, where there are multiple instances of OpenHands that could serve requests.

The cleanup probably won't quite work as-is either--we might have different SIDs in sessions than we do in conversations

I think we need to rethink this

Reply:
I think in order to rethink this, we would need to look at the amount of work being done in the runtlime_cls constructor. At the moment, it is a lot.

Reply:
oh whoops, I lost track of this: https://github.com/All-Hands-AI/OpenHands/pull/4410